### PR TITLE
fixes an issue where if a package had multiple port mappings, only th…

### PR DESCRIPTION
**Pull request by ProofOfKeags**
…e last one would get an nginx config

This ultimately will address #1045

### CODE REVIEW

File: appmgr/src/context/rpc.rs [hunk 1]
```diff
@@ -301,6 +301,10 @@ impl RpcContext {
     #[instrument(skip(self))]
     pub async fn cleanup(&self) -> Result<(), Error> {
         let mut db = self.db.handle();
+        crate::db::DatabaseModel::new()
+            .package_data()
+            .lock(&mut db, LockType::Write)
+            .await?;
```
Comment (ProofOfKeags) [inline appmgr/src/context/rpc.rs:307]: I compiled everything with `unstable` so there are some changes here related to resolving those issues.

File: appmgr/src/install/mod.rs [hunk 2]
```diff
@@ -126,7 +126,7 @@ pub async fn install(
     match pde.take() {
         Some(PackageDataEntry::Installed {
             installed,
-            manifest,
+            manifest: _manifest,
```
Comment (ProofOfKeags) [inline appmgr/src/install/mod.rs:129]: this is to quiet a warning. I made changes in every file I touched for other reasons to take warnings to 0 in those modules.

File: appmgr/src/net/nginx.rs [hunk 3]
```diff
@@ -150,7 +152,7 @@ impl NginxControllerInner {
         match self.interfaces.get_mut(&package) {
             None => {
                 let info = PackageNetInfo {
-                    ip: ipv4,
+                    _ip: ipv4,
```
Comment (ProofOfKeags) [inline appmgr/src/net/nginx.rs:155]: quiet a warning

File: appmgr/src/net/nginx.rs [hunk 4]
```diff
@@ -214,7 +216,7 @@ impl NginxControllerInner {
     }
 }
 struct PackageNetInfo {
-    ip: Ipv4Addr,
+    _ip: Ipv4Addr,
```
Comment (ProofOfKeags) [inline appmgr/src/net/nginx.rs:219]: quiet a warning

File: appmgr/src/marketplace.rs [hunk 5]
```diff
@@ -24,12 +28,16 @@ pub async fn set_eos_url(#[context] ctx: RpcContext, #[arg] url: Url) -> Result<
 pub async fn set_package_url(#[context] ctx: RpcContext, #[arg] url: Url) -> Result<(), Error> {
     let mut db = ctx.db.handle();
     let mut tx = db.begin().await?;
-    ctx.set_nginx_conf(&mut tx).await?;
+    crate::db::DatabaseModel::new()
+        .server_info()
+        .lock(&mut tx, LockType::Write)
+        .await?;
```
Comment (ProofOfKeags) [inline appmgr/src/marketplace.rs:34]: fix locking issues

File: appmgr/src/marketplace.rs [hunk 6]
```diff
@@ -10,6 +10,10 @@ use crate::Error;
 pub async fn set_eos_url(#[context] ctx: RpcContext, #[arg] url: Url) -> Result<(), Error> {
     let mut db = ctx.db.handle();
     let mut tx = db.begin().await?;
+    crate::db::DatabaseModel::new()
+        .server_info()
+        .lock(&mut tx, LockType::Write)
+        .await?;
```
Comment (ProofOfKeags) [inline appmgr/src/marketplace.rs:16]: fix locking issues

File: appmgr/src/install/mod.rs [hunk 7]
```diff
@@ -1020,13 +1020,19 @@ pub async fn install_s9pk<R: AsyncRead + AsyncSeek + Unpin>(
...
-        .get(&mut tx, true)
-        .await?
-        .into_owned()
-    {
+    } else if let Some(recovered) = {
+        // solve taxonomy escalation
+        crate::db::DatabaseModel::new()
+            .recovered_packages()
+            .lock(&mut tx, LockType::Write)
+            .await?;
```
Comment (ProofOfKeags) [inline appmgr/src/install/mod.rs:1028]: fix locking issues

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**